model/Index/Service/AdvancedSearchIndexDocumentBuilder.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoAdvancedSearch\model\Index\Service;
+
+use common_Exception;
+use common_exception_InconsistentData;
+use core_kernel_classes_Resource;
+use oat\oatbox\service\ServiceManager;
+use oat\tao\model\search\index\DocumentBuilder\IndexDocumentBuilderInterface;
+use oat\tao\model\search\index\IndexDocument;
+use oat\tao\model\search\index\IndexService;
+use oat\tao\model\TaoOntology;
+use oat\taoMediaManager\model\relation\service\IdDiscoverService;
+use oat\taoQtiItem\model\qti\Img;
+use oat\taoQtiItem\model\qti\parser\ElementReferencesExtractor;
+use oat\taoQtiItem\model\qti\QtiObject;
+use oat\taoQtiItem\model\qti\Service as QtiItemService;
+use oat\taoQtiItem\model\qti\XInclude;
+use Psr\Container\ContainerInterface;
+use ReflectionProperty;
+use taoQtiTest_models_classes_QtiTestService as QtiTestService;
+use Exception;
+use taoQtiTest_models_classes_QtiTestServiceException;
+
+class AdvancedSearchIndexDocumentBuilder implements IndexDocumentBuilderInterface
+{
+    public const QTI_IDENTIFIER_KEY = 'qit_identifier';
+    public const REFERENCES_KEY = 'referenced_resources';
+    public const ASSETS_KEY = 'asset_uris';
+    public const ITEMS_KEY = 'item_uris';
```

Comment:
We can remove the constant not used and make them private if no used elsewhere.

```suggestion
    private const QTI_IDENTIFIER_KEY = 'qit_identifier';
    private const ASSETS_KEY = 'asset_uris';
    private const ITEMS_KEY = 'item_uris';
```

Reply:
Done 69b452f :+1: